app/frontend/schools/coaches/components/SA_Coaches_SchoolIndex.res

```diff
@@ -12,48 +12,111 @@ type formVisible =
+        state.coaches,
+      )->List.length
+    {...state, coachCount: count}
   }
 
 @react.component
 let make = (~coaches, ~authenticityToken) => {
-  let (state, send) = React.useReducer(reducer, {coaches: coaches, formVisible: None})
+  let (state, send) = React.useReducer(
+    reducer,
+    {coaches: coaches, formVisible: None, coachCount: Belt.List.length(coaches)},
+  )
+
+  let currentPath = "coaches?status="
+  let isArchivedTabSelected = isSelectedTabArchived()
 
   let closeFormCB = () => send(UpdateFormVisible(None))
-  let updateCoachCB = coach => send(UpdateCoaches(coach))
+  let updateCoachCB = coach => {
+    send(UpdateCoaches(coach))
+    send(UpdateCoachCount(isArchivedTabSelected))
+  }
   <div role="main" className="flex min-h-full bg-gray-50">
     {switch state.formVisible {
     | None => React.null
     | CoachEditor(coach) =>
       <SA_Coaches_CoachEditor coach closeFormCB updateCoachCB authenticityToken />
     }}
     <div className="flex-1 flex flex-col">
-      <div className="flex px-6 py-2 items-center justify-between">
-        <button
-          onClick={_event => {
-            ReactEvent.Mouse.preventDefault(_event)
-            send(UpdateFormVisible(CoachEditor(None)))
-          }}
-          className="max-w-2xl w-full flex mx-auto items-center justify-center relative bg-white text-primary-500 hover:text-primary-600 hover:shadow-lg focus:outline-none focus:shadow-lg focus:border-primary-300 focus:text-primary-600 border-2 border-primary-300 border-dashed hover:border-primary-300 p-6 rounded-lg mt-8 cursor-pointer">
-          <i className="fas fa-plus-circle text-lg" />
-          <h5 className="font-semibold ms-2"> {tr("add_new_coach") |> str} </h5>
-        </button>
+      <div className="w-full pt-2 relative md:sticky top-0 z-20 bg-gray-50 border-b">
+        <div className="max-w-3xl mx-auto">
+          <div className="px-12 py-5 flex justify-between items-baseline">
+            <h2 className="font-bold ms-2 text-xl"> {tr("coaches")->str} </h2>
+            <button
+              onClick={_event => {
+                ReactEvent.Mouse.preventDefault(_event)
+                send(UpdateFormVisible(CoachEditor(None)))
+              }}
+              className="flex items-center bg-primary-500 text-white py-3 px-6 justify-between rounded-lg text-primary-500 hover:text-white hover:shadow-lg focus:outline-none focus:shadow-lg focus:border-primary-300 focus:text-white">
+              <i className="fas fa-plus-circle text-lg" />
+              <h5 className="ms-2 text-sm font-medium"> {tr("add_new_coach") |> str} </h5>
+            </button>
+          </div>
+        </div>
+        <div className="max-w-3xl mx-auto">
+          <div className="px-12 flex justify-start ">
+            <div
+              className={`px-5 py-2 flex items-center p-2 font-medium
+               ${!isArchivedTabSelected ? "border-b-2 border-primary-500 text-primary-500" : ""}`}>
```

Comment:
This shouldn't be the `flex` container, rather `a` should be the flex container.